packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -402,31 +407,90 @@ const SideNav = ({
     [isCompact],
   );
 
-  const processedSideNavItems = useMemo(
-    () =>
-      sideNavItems.map((item) => ({
-        ...item,
-        children: item.children?.map((childProps) => {
-          return (
+  const setSelectedInChildItems = useCallback(
+    ({
+      nestedNavItems,
+      selectedItemId,
+    }: {
+      nestedNavItems: Array<Omit<SideNavItem, "startIcon" | "endIcon">>;
+      selectedItemId: string;
+    }) => {
+      const childItems = nestedNavItems.map((childItem) => {
+        if (childItem.isSelected) {
+          childItem.isSelected = false;
+        }
+        if (childItem.id === selectedItemId) {
+          childItem.isSelected = true;
+        }
+        if (childItem.nestedNavItems) {
+          setSelectedInChildItems({
+            nestedNavItems: childItem.nestedNavItems,
+            selectedItemId,
+          });
+        }
+        return childItem;
+      });
+      return childItems;
+    },
+    [],
+  );
+
+  const setSelectedItem = useCallback(
+    (selectedItemId: string) => {
+      const newNavItems = sideNavItemsList.map((item) => {
+        if (item.isSelected) {
+          item.isSelected = false;
+        }
+        if (item.id === selectedItemId) {
+          item.isSelected = true;
+        }
+        if (item.nestedNavItems) {
+          item.nestedNavItems = setSelectedInChildItems({
+            nestedNavItems: item.nestedNavItems,
+            selectedItemId,
+          });
+        }
+        return item;
+      });
+      updateSideNavItemsList(newNavItems);
+    },
+    [sideNavItemsList, setSelectedInChildItems],
+  );
+
+  const processedSideNavItems = useMemo(() => {
+    return sideNavItemsList?.map((item) => ({
+      ...item,
+      childNavItems: item.nestedNavItems?.map((childProps) => {
+        return {
+          id: childProps.id,
+          isSelected: childProps.isSelected,
+          isDisabled: childProps.isDisabled,
+          navItem: (
             <SideNavItemContentContext.Provider
-              value={{ ...sideNavItemContentProviderValue, depth: 2 }}
+              value={{
+                ...sideNavItemContentProviderValue,
+                depth: 2,
+                isSortable: item.isSortable,
+              }}
               key={childProps.id}
             >
               <SideNavItemContent
                 {...childProps}
                 key={childProps.id}
```

Comment:
We don't need `key` passed to both of these. `key` is an `id` React can use to track that exact component in an array such that if it moves around, they don't re-render the whole list, they just move that one element to a new position in the list.